### ISSUE TITLE
Sanity checks for the data

### ISSUE THREAD

**Issue by ohsOllila** [NMRLipids/Databank#3]
Need for various sanity checks for the databank has been discussed several times during the project. In online meeting on 26th of February 2021, we decided to compare the total number of atoms between the databank information and actual simulation. 

Also other possible sanity checks were mentioned, but concrete decisions were not made. We can continue the discussion here.

**Comment by ohsOllila** [comment]
Comparison between the total number of atoms in the databank information and actual simulation is now added to the AddData.py. If these do not match, the code stops. This is currently based on number of atoms given in the mapping file. However, the atomnames in the mapping file are not checked. Therefore, it is still possible to give a mapping file with incorrect atom names as long as the number of lines is correct.

**Comment by batukav** [comment]
I think building this check on the mapping file only could create a problem with the polarizable force field simulations. For the Drude force field the Drude particles won't show up in the mapping file so there will always be a mismatch between the number of atoms in the trajectory and the mapping file. 

One possible work around is to select the atoms in the trajectory by mass (mass > 0.25 amu, I think) as these Drude particles have a very low mass by definition. However, I don't know how robust this solution for the other polarizable force fields.

**Comment by ohsOllila** [comment]
Maybe one option would be to put the drude particles into the mapping file for the polarizable force fields?

**Comment by batukav** [comment]
> Maybe one option would be to put the drude particles into the mapping files for the polarizable force fields?

Yes I think this is a rather straightforward way to go for. In this case we might need to modify how the order parameter script handles the mapping file, though, as we should make sure that the drude pairs won't be passed into the order parameter script.

Regarding this overall sanity check, at the moment, can we handle if only a part of the total system is uploaded? For instance, if someone uploads the lipids only, or lipids + ions only, will our sanity check still work?

**Comment by ohsOllila** [comment]
>Regarding this overall sanity check, at the moment, can we handle if only a part of the total system is uploaded? For instance, if >someone uploads the lipids only, or lipids + ions only, will our sanity check still work?

I think that this should work, but I have not tried.

**Comment by ohsOllila** [comment]
> Comparison between the total number of atoms in the databank information and actual simulation is now added to the AddData.py. If these do not match, the code stops.

Because there are situations in united atom models where individual hydrogens are present explicitly (for example for POPS), we had to add a possibility to bypass this test in AddData.py. However, this should be done only when it is absolutely clear that mismatch in atom numbers is not a real problem.

Otherwise, we have added 'WARNING:' keys to the README.yaml for the cases when there is something inconvenient the trajectory, such as membrane is in atypical orientation, old Gromacs version is used, or some atoms do not have unique names. However, these are not checked automatically but have to included manually. The simulations with problematic warnings can be then skipped when analyzing the databank. This is not yet described in the [instructions](https://github.com/NMRLipids/Databank/blob/main/Scripts/BuildDatabank/info_files/README.md). 

I would be happy to hear more ideas about sanity checks in the future, but I would make the first publication with this setup.
